From: jbeulich@suse.com
Subject: MSI-X: fix an error path
Patch-mainline: Never, SUSE-Xen specific
References: luckily none so far

The wrong variable was used as array index during the original forward
porting process.

--- a/drivers/pci/msi-xen.c
+++ b/drivers/pci/msi-xen.c
@@ -599,7 +599,7 @@ static int msix_capability_init(struct p
 		int avail = i - 1;
 		for (j = --i; j >= 0; j--) {
 			list_for_each_entry(pirq_entry, &dev->msi_list, list)
-				if (pirq_entry->entry_nr == entries[i].entry)
+				if (pirq_entry->entry_nr == entries[j].entry)
 					break;
 			msi_unmap_pirq(dev, entries[j].vector,
 				       msi_dev_entry->owner,
